src/errno.rs

```diff
@@ -24,9 +24,9 @@ fn errno_location() -> *mut c_int {
 }
 
 #[cfg(target_os = "dragonfly")]
-unsafe fn errno_location() -> *mut c_int {
-    extern { fn __dfly_error() -> *mut c_int; }
-    __dfly_error()
+#[link(name="errno_dragonfly", kind="static")]
```

Comment:
I thought this was thread-local. Do we just not use that yet because Rust doesn't support it on stable yet?

Reply:
Exactly. And it's unclear when this will become stable. The suggestion to use a C extension for this purpose came from @alexcrichton in this thread https://github.com/rust-lang/rust/issues/29594#issuecomment-314115129